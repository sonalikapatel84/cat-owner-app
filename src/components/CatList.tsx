import React from 'react'

interface CatListProps {
  gender: string
  cats: string[]
}

const CatList: React.FC<CatListProps> = ({ gender, cats }) => (
  <div>
    <h2>{gender}</h2>
    <ul>
      {cats.map((cat) => (
        <li key={cat}>{cat}</li>
      ))}
    </ul>
  </div>
)

export default CatList
