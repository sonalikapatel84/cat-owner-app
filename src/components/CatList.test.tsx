import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CatList from './CatList'

describe('CatList Component', () => {
  // Test rendering with provided props
  test('renders the gender and list of cats', () => {
    const gender = 'Male'
    const cats = ['Garfield', 'Tom']

    render(<CatList gender={gender} cats={cats} />)

    // Check if the gender is displayed
    expect(screen.getByText(gender)).toBeInTheDocument()

    // Check if each cat name is displayed
    cats.forEach((cat) => {
      expect(screen.getByText(cat)).toBeInTheDocument()
    })
  })

  // Test rendering with an empty list of cats
  test('renders without crashing when there are no cats', () => {
    const gender = 'Female'
    const cats: string[] = []

    render(<CatList gender={gender} cats={cats} />)

    // Check if the gender is displayed
    expect(screen.getByText(gender)).toBeInTheDocument()

    // Check if no cats are displayed
    expect(screen.queryByRole('list')).toBeEmptyDOMElement()
  })

  // Test snapshot rendering
  test('matches the snapshot', () => {
    const gender = 'Male'
    const cats = ['Garfield', 'Tom']

    const { asFragment } = render(<CatList gender={gender} cats={cats} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
