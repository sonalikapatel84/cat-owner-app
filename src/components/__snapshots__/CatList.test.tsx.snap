// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatList Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Male
    </h2>
    <ul>
      <li>
        Garfield
      </li>
      <li>
        Tom
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
